lib/pure/options.nim

```diff
@@ -139,6 +266,12 @@ proc get*[T](self: Option[T], otherwise: T): T =
   else:
     otherwise
 
+template either*(self, otherwise: untyped): untyped =
```

Comment:
Since this takes an untyped statement, does it follow short-circuit logic? That is, if `self` is none, `otherwise` won't be evaluated?